Tests/SwiftSyntaxBuilderTest/IfConfigDeclSyntaxTests.swift

```diff
@@ -0,0 +1,70 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import XCTest
+import SwiftSyntax
+import SwiftSyntaxBuilder
+
+final class IfConfigDeclSyntaxTests: XCTestCase {
+  func testIfConfigClauseSyntax() {
+    let buildable = IfConfigDeclSyntax(
+      clauses: IfConfigClauseListSyntax {
+        IfConfigClauseSyntax(
+          poundKeyword: .poundIfKeyword(),
+          condition: ExprSyntax("DEBUG"),
+          elements: .statements(
+            CodeBlockItemListSyntax {
+              DeclSyntax(
+                """
+                public func debug(_ data: Foo) -> String {
+                  return data.debugDescription
+                }
+                """
+              )
+            }
+          )
+        )
+        IfConfigClauseSyntax(
+          poundKeyword: .poundElseKeyword(leadingTrivia: .newline),
+          elements: .statements(
+            CodeBlockItemListSyntax {
+              DeclSyntax(
+                """
+                public func debug(_ data: Foo) -> String {
+                  return data.description
+                }
+                """
+              )
+            }
+          )
+        )
+      },
+      poundEndif: .poundEndifKeyword(leadingTrivia: .newline)
+    )
+
+    AssertBuildResult(
+      buildable,
+      """
+
+      #if DEBUG
```

Comment:
The only "downside" is that it also counts for the `#if` and not only `#elseIf`

Reply:
This should probably be fixed if you add a check to `shouldOmitNewline` that returns `true` if to token is the first in the source file, i.e. if `node.previousToken(view: .sourceAccurate) == nil`